src/lib/storage/storage_manager.hpp

```diff
@@ -62,14 +64,15 @@ class StorageManager : public Noncopyable {
   StorageManager() = default;
   friend class Hyrise;
 
-  // Tables can currently not be modified concurrently
-  std::map<std::string, std::shared_ptr<Table>> _tables;
-
-  // The map of views is locked because views are created dynamically, e.g., in TPC-H 15
-  std::map<std::string, std::shared_ptr<LQPView>> _views;
-  mutable std::unique_ptr<std::shared_mutex> _view_mutex = std::make_unique<std::shared_mutex>();
+  // We preallocate maps to prevent costly re-allocation.
+  static constexpr size_t _INITIAL_MAP_SIZE = 100;
 
-  std::map<std::string, std::shared_ptr<PreparedPlan>> _prepared_plans;
+  tbb::concurrent_unordered_map<std::string, std::shared_ptr<Table>> _tables{
+      _INITIAL_MAP_SIZE, tbb::tbb_allocator<std::pair<const std::string, std::shared_ptr<Table>>>()};
```

Comment:
Why do we need the tbb_allocator here? Anything about the map that I am not aware of?

Reply:
To be honest: I used the one from the examples, I thought it might have stronger guarantees. Also, preallocating a map without specifying an allocator is not possible.

Reply:
Just saw that this wasn't part of the template parameter, so it was used anyway. I'm surprised to see that the first constructor does not work:

https://software.intel.com/en-us/node/506172

Anyway, `{}` instead of talking about the allocator should be sufficient and keeps people from wondering.

Reply:
I think it is safe to say now that the first constructor does not work if the developer is an idiot.